### ISSUE TITLE
Composed field level validation

### ISSUE THREAD

**Issue by dmtrKovalenko** [jaredpalmer/formik#682]
**Is your feature request related to a problem? Please describe.**

Formik is really cool. I really like field-level validation. But it's missing really useful feature - multiple field level validation. 
Imagine you have to create an a required email field. You want to make something like

```jsx
<Field
  validate={[validateRequired, validateEmail]} // they are imported from somewhere
/>
```

But you need to compose them by yourself. BTW redux-form has this feature out of the box, you just need to pass an array of validators.

**Describe the solution you'd like**

I have created my own composer for multiple validation 
```ts
export const composeValidators = (...args: Array<(value: any) => string | undefined>) => (value: any) => {
  for (const validator of args) {
    const error = validator(value);

    if (error) {
      return error;
    }
  }

  return undefined;
};
```

Then in the code
```jsx
   <Field
     name="email"
     type="email"
     label={<Translate> email </Translate>}
     fullWidth
     component={TextField}
     validate={composeValidators(required, email)}
   />
```

But want to get this out of the box. 

**Additional context**

I would like to open a PR if this feature is interesting :)

**Comment by dmtrKovalenko** [comment]
@jaredpalmer @jared-ross  A lot of people pinged me out that found this proposal really useful. Can you give some feedback here for me to be clear should I submit a PR or not. Thanks so much for your awesome project! 💪

**Comment by phanhoangloc** [comment]
I have used `redux-form` but then switched to `formik` for almost 1 year because of performance reasons.

However, I still like how to configure validation rules for form fields in `redux-form`. Therefore, I created [formik-validators](https://github.com/phanhoangloc/formik-validators) and used it in some projects. 

It may not be exactly what you want because it only works with `withFormik` HOC. But I hope it can help.
